debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnector.java

```diff
@@ -60,6 +60,12 @@ public List<Map<String, String>> taskConfigs(int maxTasks) {
 
         final SqlServerConnectorConfig config = new SqlServerConnectorConfig(Configuration.from(properties));
 
+        if (maxTasks > 1 && properties.containsKey(DATABASE_NAMES.name()) && config.getDatabaseNames().size() < maxTasks) {
+            LOGGER.info(String.format("maxTasks: %d is greater than the number of databases: %d, overriding maxTasks to %d", maxTasks, config.getDatabaseNames().size(),
```

Comment:
I don't think it should be done here and requires overriding the config value and logging (it's not really something special). Instead, `buildTaskConfigs()` could use `min(maxTasks, config.getDatabaseNames().size())` instead of `maxTasks` as the size of `databasesByTask`.

Reply:
> While the connector validates the database.names parameter, the task doesn't. It should validate it [...]
It looks like validation is done here in `BaseSourceTask`: https://github.com/sugarcrm/debezium/blob/260462b1a2a209f16c40f2f85f55a2d8c0aef0a3/debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java#L128-#L130 but the `SQLServerConnectorConfig` was missing a check for an empty string. I've added a commit to validate the condition and corresponding test.